Rude/Rude.cs

```diff
@@ -8,208 +8,77 @@
-            bool kick = false;
-            lock (Rudes)
-            {
-                var oneYear = DateTimeOffset.Now.AddYears(-1);
-                var actorOutRudeQuery = Rudes
-                        // get all rudes which actor has done
-                        .Where(r => r.ActorId == actor.UserId)
-                        // and where he didn't rude himself by ruding too often
-                        .Where(r => r.ActualTargetId == null);
-
-                // if actor rudes himself or has already ruded in last Minute
-                bool punishActor = actorOutRudeQuery
-                        // take all rudes, where the actor hasn't kicked someone
-                        //  (kicking by ruding does not have a cooldown)
-                        .Where(r => r.Duration != null)
-                        // test, if the actor has ruded anyone in last minute
-                        .Any(r => r.Timestamp > DateTime.Now.AddMinutes(-1));
-
-                // if actor already ruded in last minute, he will be ruded instead of target
-                if (punishActor)
-                {
-                    rudeEntity.ActualTargetId = rudeEntity.TargetId;
-                    rudeEntity.TargetId = actor.UserId;
-                    // he will be the new target
-                    target = actor;
-                }
-
-                // get active rudes on target and add current rude
-                targetInRudes = GetActiveInRudes(target.UserId).Count() + 1;
-                if (targetInRudes > PREFIXES.Length && !punishActor)
-                {
-                    kick = true;
-                    rudeEntity.Duration = null;
-                }
-                else
-                {
-                    var actorOutRudes = actorOutRudeQuery.Count();
-                    var medianQuery =
-                            from r in Rudes
-                            where r.Timestamp > oneYear
-                            group Math.Log((r.Timestamp - oneYear).TotalMinutes) by r.ActorId into g
-                            select g.Sum();
-                    var median = medianQuery.Median();
-
-                    double durationFactor = actorOutRudes == 0 ? 2 : median.Value / actorOutRudes;
-                    durationFactor = Math.Max(durationFactor, 0.25);
-                    durationFactor = Math.Min(durationFactor, 2);
-                    durationFactor *= rng.NextDouble() * targetInRudes;
-                    var duration = TimeSpan.FromHours(durationFactor * 2).Ticks;
-                    rudeEntity.Duration = TimeSpan.FromTicks(Math.Min(duration, TimeSpan.FromDays(1).Ticks));
-                }
-
-                // add rude
-                Rudes.Add(rudeEntity);
-            }
+            rude.RaiseRudeLevel();
 
-            if (kick)
+            if (rude.RudeLevel > Prefixes.Length)
             {
-                await target.Kick(REASONS[rng.Next(REASONS.Length)]);
+                RudeStatus.Remove(actor.UserId);
```

Comment:
Perhaps we should do away with kicking entirely. Now that auto-reconnect is gone it can be *really* annoying (especially if it happens while you're afk and then you spend the rest of the day wondering why no one is there).